services/vpm/vpm-version.service.js

```diff
@@ -0,0 +1,81 @@
+import Joi from 'joi'
+import { optionalUrl } from '../validators.js'
+import { latest, renderVersionBadge } from '../version.js'
+import { BaseJsonService, NotFound } from '../index.js'
+
+const queryParamSchema = Joi.object({
+  repositoryUrl: optionalUrl.required(),
+  sort: Joi.string().valid('defined', 'semver').default('defined'),
+}).required()
+
+const schema = Joi.object({
+  packages: Joi.object()
+    .pattern(
+      /./,
+      Joi.object({
+        versions: Joi.object().pattern(/./, Joi.object()).required(),
+      }).required()
+    )
+    .required(),
+}).required()
+
+export default class VpmVersion extends BaseJsonService {
+  static category = 'version'
+
+  static route = {
+    base: 'vpm/v',
+    pattern: ':packageId',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'VPM Package Version',
+      namedParams: {
+        packageId: 'com.vrchat.udonsharp',
+      },
+      queryParams: {
+        repositoryUrl: 'https://packages.vrchat.com/curated?download',
```

Comment:
I realise that this is a user-supplied parameter so the user can pass in whatever they want, but for a version badge it seems like you could pass in https://packages.vrchat.com/curated (without `?download`) here and works just the same. For the purposes of the example/test, do we need this?

Reply:
The reason why I use url with `? download` os it's the url vpm uses.